File: Javascript/CorrecaoHeranca/src/preguica.ts
import { Animal } from "./animal";

export class Preguica implements Animal{

    nome: string;
    idade: number;

    constructor(
        nome: string,
        idade: number
    ){
        this.nome = nome
        this.idade = idade
    }

    emitirSom(): void {
        console.log("Sons de Preguiça")
    }
    locomocao(): void {
        console.log("Subindo em árvores")
    }

}

File: Kotlin/HerancaPoo/src/main/kotlin/exercicio/Preguica.kt
package exercicio

class Preguica(
    nome: String, idade: Int
) : Animal(nome, idade){

    override fun emitirSom() {
        println("Voz do Sid")
    }

    override fun locomocao() {
        println("Subindo em Árvores")
    }

}